### PR TITLE
Implements to_html for Dataset

### LINKED ISSUE #496: _repr_html_ for improved rendering in Jupyter notebooks
The current text output from `__repr__` is too messy due to the inclusion of arrays of values and variances. This will only get worse when we have larger datasets with more items.

A quick solution would be to show values and variances only on hover, using some css.

It turns out that xarray is working on something much better: https://github.com/pydata/xarray/pull/1820. See https://jsfiddle.net/tay08cn9/4/ for a preview. Can we benefit from this development and reuse/adapt it, or have some common code?

### PR DESCRIPTION
**Pull request by dtasev**
Uses xarray's excellent HTML table (https://github.com/pydata/xarray/pull/3425) as a starting point and builds a few things on top:

- Makes it compatible with scipp.Dataset. This breaks any xarray compatibility. The two could be merged, but for this PR it was not deemed worthwhile.
- New sections added to the display: Labels, Masks. 
- Displays Variances, and Value/Variance column headers

Other functionality should be the same as the original xarray implementation.

DataArray will be covered in a follow-up PR.

Closes #496 

To use:

```
from scipp.table_html import to_html
                                    
to_html(dataset)
```

Some examples I used:
```
import scipp as sc
import numpy as np
from scipp import Dim

var = sc.Variable([Dim.X], values=np.arange(4))
sparse = sc.Variable([sc.Dim.Y], [sc.Dimensions.Sparse])

ds = sc.Dataset({
            'a': sc.Variable([Dim.X, Dim.Y], unit=sc.units.m, values=np.arange(10).reshape(5,2), variances=np.arange(10).reshape(5,2)),
            'b':sc.Variable([Dim.Z], values=np.arange(12)),
            'c':sc.Variable([Dim.Z], values=np.arange(12), variances=np.arange(12)),
            'd':sc.Variable([Dim.Z], values=np.arange(12, dtype=np.double), variances=np.arange(12, dtype=np.double)),
                       
    },
        coords={
            Dim.X: sc.Variable([Dim.X, Dim.Y], values=np.arange(10).reshape(5,2)),
            Dim.Z:sc.Variable([Dim.Z], values=np.arange(12)),
            Dim.Row: sc.Variable([Dim.X], values=np.arange(0, 8, 2)),
        },
        attrs={
        "number_of_potatoes": sc.Variable([Dim.X, Dim.Y], values=np.arange(10).reshape(5,2)),
        "title": sc.Variable("Test Dataset"),
        "scalar": sc.Variable(1)
        })
ds["a"].attrs["h"] = sc.Variable([Dim.X, Dim.Y], values=np.arange(10).reshape(5,2))


from scipp.table_html import to_html
                                    
to_html(ds)
```

With sparse data:
```
import scipp as sc
import numpy as np
from scipp import Dim

var = sc.Variable([Dim.X], values=np.arange(4))
sparse = sc.Variable([Dim.X, Dim.Y], [4, sc.Dimensions.Sparse], variances=True)

sparse[Dim.X, 0].values = np.arange(12)
sparse[Dim.X, 1].values = np.arange(2)
sparse[Dim.X, 2].values = np.arange(44)
sparse[Dim.X, 3].values = np.arange(3)

sparse[Dim.X, 0].variances = np.arange(12)
sparse[Dim.X, 1].variances = np.arange(2)
sparse[Dim.X, 2].variances = np.arange(44)
sparse[Dim.X, 3].variances = np.arange(3)


ds_sparse = sc.Dataset({
            'sparse':sparse,
    },
    coords={
        Dim.X: sc.Variable([Dim.X], values=np.arange(5))
    },
    labels={"mon": sc.Variable([Dim.X], values=np.array([1,1,5,5]))},
      masks={"mask": sc.Variable([Dim.X], values=np.array([True,False,True,False]))},
    attrs={"test": sc.Variable([Dim.X], values=np.random.rand(5))})

from scipp.table_html import to_html
                                    
to_html(ds_sparse)
```

### CODE REVIEW

File: python/src/scipp/table_html/style.css [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+  text-align: left;
+  color: #222;
+}
+.xr-value-preview {
+  grid-column: 5;
+}
+.xr-variance-preview {
+  grid-column: 6;
+}
+.xr-var-preview-variances{
```
Comment (dtasev) [inline python/src/scipp/table_html/style.css]: This class is an exception of replacing `xr` with `sc` for scipp-specific classes, but I thought it made more sense to keep it next to the other CSS definitions that format the general grid layout

File: python/src/scipp/table_html/formatting_html.py [hunk 2]
```diff
@@ -0,0 +1,396 @@
...
+def summarize_variable(name, var, is_index=False, is_attr=None):
+    """
+    :param name:
+    :param var:
+    :param is_index: If the variable is an index - used to bold
+                     coordinates that represent the indices of
+                     a dimension that the data contains
+
+    :param is_attr: If the variable is for an attribute, then this
+                    hides the show/hide attributes button.
```
Comment (SimonHeybrock) [inline python/src/scipp/table_html/formatting_html.py]: I think xarray stores attributes with variables, but for scipp's variables we cannot have attributes. `DataArray` has the attributes in our case. That is, I think, we should hide the button also for coords, labels, masks.

File: python/src/scipp/table_html/formatting_html.py [hunk 3]
```diff
@@ -0,0 +1,396 @@
...
+        disabled = "disabled"
+        attrs_ul = None
+
+    preview = inline_variable_repr(var)
+    data_repr = f"Values:<br>{short_data_repr_html(var)}"
+    variances_preview = ""
+    if var.variances is not None:
+        variances_preview = inline_variable_repr(var, has_variances=True)
+        data_repr += f"<br><br>Variances:<br>\
+                       <div>{short_data_repr_html(var)}</div>"
```
Comment (SimonHeybrock) [inline python/src/scipp/table_html/formatting_html.py]: Somehow there is an extra empty line after "Variances:" which is not there after "Values:". Is it due to the `<div>`?

**Comment by SimonHeybrock** [comment]
Since there is probably more to test, and a number of issues that can be fixed as follow-ups (potentially in parallel by several contributors) I will merge this as-is. It is a new stand-alone feature so it does not risk breaking anything.

Btw: It also works on the generated Sphinx pages. :)

File: python/src/scipp/table_html/formatting_html.py (hunk 3 above)
Comment (dtasev) [reply python/src/scipp/table_html/formatting_html.py]: I cannot see it, could you post a screenshot if possible?

**Pull request merged**